### ISSUE TITLE
Sub folders and pdoc without __init__.py files

### ISSUE THREAD

**Issue by ferreteleco** [mitmproxy/pdoc#405]
#### Problem Description
Hi all! I've just discovered this awesome package and was giving it a try when I found something that doesn't fit (for me at least): I'm working on a project with many sub-folders inside without using __init__.py other than in the top level one. When I trigger pdoc, none of the sub-modules get documented. Including empty __init__.py files in each sub folder solves the problem, but I found it quite odd (since python does not need that in order to work properly).

Thanks in advance! 
#### Steps to reproduce the behavior:
1. Project with many sub-folders without __init__.py files
2. Triggering pdoc does not generate documentation for files un those folders
3. Including __init__.py files in each sub folder solves the problem

#### System Information
pdoc: 12.0.2
Python: 3.10.2
Platform: Linux-5.13.0-48-generic-x86_64-with-glibc2.31

**Comment by mhils** [comment]
The [Python documentation](https://docs.python.org/3/tutorial/modules.html#packages) states that `__init__.py` files are required:

> The `__init__.py` files are required to make Python treat directories containing the file as packages.

A package directory without `__init__.py` is a [PEP 420](https://www.python.org/dev/peps/pep-0420/) "namespace package", not a regular package. See also https://dev.to/methane/don-t-omit-init-py-3hga, which is written by one of the CPython Core Developers.

I agree it's all a bit confusing, but for pdoc it boils down to the fact that [`pkgutil.walk_packages`](https://docs.python.org/3/library/pkgutil.html#pkgutil.walk_packages) - which we use to discover packages - does not pick up namespace packages. See also: https://github.com/python/cpython/issues/73444

Long story short, there's not much we can do here. I'd really recommend to add `__init__.py` files. As a workaround, you could maybe list all the sub-packages explicitly. Hope this helps!
